### PR TITLE
Prevent repetition answer

### LINKED ISSUE #18: Prevent answers to be selected more than once.
When an answer is clicked more than once, and it's correct, the score keeps incrementing, and this shouldn't be the case.

### PR DESCRIPTION
**Pull request by AsiyeGokalp**
Closes #18

### CODE REVIEW

File: src/pages/questionPage.js [hunk 1]
```diff
@@ -31,7 +31,9 @@ export const initQuestionPage = () => {
     const correctAnswer = currentQuestion.correct;
     if (selectedAnswer.innerText[0] === correctAnswer) {
       selectedAnswer.classList.add('correct');
-      quizData.finalScore++;
+      if (currentQuestion['selected'] === null) {
```
Comment (arkaitzgarro) [inline src/pages/questionPage.js:34]: It is a micro optimisation, but if you check if the question has been answered at the very beginning of the function, you can avoid the rest of the code to be executed.

Comment (arkaitzgarro) [reply src/pages/questionPage.js:34]: It also makes the code cleaner, because nested if statements are hard to read at some point.

Comment (arkaitzgarro) [reply src/pages/questionPage.js:34]: https://gomakethings.com/the-early-return-pattern-in-javascript/

**Comment by AsiyeGokalp** [comment]
Can 'early return pattern' be used for if/else, I mean if 'if' has an else part can we convert to this type?

File: src/pages/questionPage.js [hunk 2]
```diff
@@ -31,12 +31,14 @@ export const initQuestionPage = () => {
     const correctAnswer = currentQuestion.correct;
     if (selectedAnswer.innerText[0] === correctAnswer) {
       selectedAnswer.classList.add('correct');
+      if (!(currentQuestion['selected'] === null)) return;
```
Comment (arkaitzgarro) [inline src/pages/questionPage.js:34]: The idea of early return is to exit as soon as soon as possible. The condition is correct, but this needs to happen at the very beginning of the function.

**Comment by arkaitzgarro** [comment]
> Can 'early return pattern' be used for if/else, I mean if 'if' has an else part can we convert to this type?

Yes of course, you could convert something like this:

```js
if (condition) {
  // do something
} else {
  // do something else
}
```

into this:

```js
if (condition) {
 // do something
 return 
}

// do something else
```

**Comment by AsiyeGokalp** [comment]
Thank you

**Review by arkaitzgarro — APPROVED** [review]

**Pull request merged**